### PR TITLE
(feat raredisease) add upload and clinical delivery

### PR DESCRIPTION
**Pull request by peterpru**
### Added

- Upload to genotypes, locusDB and gens for raredisease workflow

### Changed

-

### Fixed

-

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test
(starting from a run and stored case on stage: amusingmarmoset)
- [x] cg upload genotypes  amusingmarmoset
- [x] cg upload observations  amusingmarmoset
- [x] cg upload gens amusingmarmoset
- [x] cg upload clinical-delivery  amusingmarmoset

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by rannick** [comment]
This is not completely finished: I want to move the location of the extraction of the sex from hk files to a common location (in metrics probably) but it is time to discuss:
- Is it worth making the duplication rate go down by aggregating MIP and raredisease, if we are going to deprecate MIP anyways?

File: cg/meta/upload/gt.py [hunk 1]
```diff
@@ -82,12 +98,23 @@ def _get_samples_sex_balsamic(self, case_obj: Case) -> dict:
...
     def analysis_sex(self, qc_metrics_file: Path) -> dict:
         """Fetch analysis sex for each sample of an analysis."""
-        qc_metrics: MIPMetricsDeliverables = self.get_parsed_qc_metrics_data(qc_metrics_file)
+        qc_metrics: MIPMetricsDeliverables = self.get_parsed_qc_metrics_data_mip(qc_metrics_file)
         return {
             sample_id_metric.sample_id: sample_id_metric.predicted_sex
             for sample_id_metric in qc_metrics.sample_id_metrics
         }
 
+    def analysis_sex_raredisease(self, qc_metrics_file: Path, sample_id: Sample) -> dict:
```
Comment (ivadym) [inline cg/meta/upload/gt.py:107]: RareDiseaseAnalysisAPI?

File: cg/meta/report/raredisease.py
Comment (ivadym) [inline cg/meta/report/raredisease.py:1]: I would leave the delivery report for another PR. I think we're trying to address too many things here that could each be worth their own PR: delivery report, observations upload, delivery to Caesar, genotype upload, scout upload 😅

**Review by ivadym — COMMENTED** [review]
🔥 💯

File: cg/constants/delivery.py [hunk 2]
```diff
@@ -150,6 +149,29 @@
...
+    {"vcf-sv-research"},
+    {"vcf-sv-research-index"},
+]
+
+RAREDISEASE_ANALYSIS_SAMPLE_TAGS: list[set[str]] = [
+    {AlignmentFileTag.BAM},
+    {AlignmentFileTag.BAM_BAI},
+    {AlignmentFileTag.CRAM},
+    {AlignmentFileTag.CRAM_INDEX},
+]
```
[the remark this answers is not in the document]
Comment (rannick) [reply cg/constants/delivery.py:173]: I don't understand the suggestions here. Do you mean: get rid of those and fetch the files tagged with clinical-delivery instead?

File: cg/constants/observations.py [hunk 3]
```diff
@@ -50,6 +50,14 @@ class MipDNAObservationsAnalysisTag(StrEnum):
     FAMILY_PED: str = "pedigree"
 
 
+class RarediseaseObservationsAnalysisTag(StrEnum):
+    """Rare disease observations files analysis tags."""
+
+    SNV_VCF: str = "vcf-snv"
```
[the remark this answers is not in the document]
Comment (rannick) [reply cg/constants/observations.py:56]: According to @jemten, they contain the same information. But it would be good to have a wider discussion about this

File: cg/meta/upload/raredisease/raredisease.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    def __init__(self, config: CGConfig):
+        self.analysis_api: RarediseaseAnalysisAPI = RarediseaseAnalysisAPI(config)
+        super().__init__(config=config, analysis_api=self.analysis_api)
+
+    def upload(self, ctx: click.Context, case: Case, restart: bool) -> None:
+        """Uploads RAREDISEASE analysis data and files."""
+        analysis: Analysis = case.analyses[0]
+        self.update_upload_started_at(analysis=analysis)
+
```
[the remark this answers is not in the document]
Comment (rannick) [reply cg/meta/upload/raredisease/raredisease.py:34]: Is done in this PR: https://github.com/Clinical-Genomics/cg/pull/3267

File: cg/constants/observations.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rannick) [reply cg/constants/observations.py:56]: (we are not producing the gbcf file with raredisease)

File: cg/meta/observations/raredisease_observations_api.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+from cg.meta.workflow.raredisease import RarediseaseAnalysisAPI
+from cg.models.cg_config import CGConfig
+from cg.models.observations.input_files import RarediseaseObservationsInputFiles
+from cg.store.models import Case
+from cg.utils.dict import get_full_path_dictionary
+
+LOG = logging.getLogger(__name__)
+
+
+class RarediseaseObservationsAPI(ObservationsAPI):
```
[the remark this answers is not in the document]
Comment (rannick) [reply cg/meta/observations/raredisease_observations_api.py:33]: Yes, that was my thinking, trying to make the break with MIP as clean as possible. However, one can argue to make ObservationAPI closer to MIPObservationAPI, do you think that would be advantageous?

File: cg/constants/delivery.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ivadym) [reply cg/constants/delivery.py:173]: Yes, exactly! What's delivered and kept for long term storage should be defined in Hermes with the `clinical-delivery` and `long-term-storage` tags

File: cg/meta/observations/raredisease_observations_api.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ivadym) [reply cg/meta/observations/raredisease_observations_api.py:33]: I don't think so... But let's make our lives easier for the future (and avoid changing the other observations' APIs in this PR)

File: cg/meta/upload/raredisease/raredisease.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (rannick) [reply cg/meta/upload/raredisease/raredisease.py:34]: Was it what you meant?

File: cg/meta/upload/gt.py [hunk 6]
```diff
@@ -105,14 +132,30 @@ def get_qcmetrics_file(self, hk_version_obj: Version) -> Path:
         LOG.debug(f"Found qc metrics file {hk_qcmetrics.full_path}")
         return Path(hk_qcmetrics.full_path)
 
+    def get_qcmetrics_file_raredisease(self, hk_version_obj: Version) -> Path:
```
[the remark this answers is not in the document]
Comment (rannick) [reply cg/meta/upload/gt.py:135]: Yes!

File: cg/meta/report/raredisease.py
Comment (rannick) [reply cg/meta/report/raredisease.py:1]: scout will be separate, but yes, it got out of hand. Do you want me to separate?

File: tests/cli/upload/test_cli_upload_auto.py
[the remark this answers is not in the document]
Comment (rannick) [reply tests/cli/upload/test_cli_upload_auto.py:1]: [b1bc817](https://github.com/Clinical-Genomics/cg/pull/3165/commits/b1bc817138dd3dabf71867cc4efe159a55712541)

File: cg/constants/delivery.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rannick) [reply cg/constants/delivery.py:173]: 181f0f46b

File: cg/meta/upload/raredisease/raredisease.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        # Main upload
+        ctx.invoke(upload_genotypes, family_id=case.internal_id, re_upload=restart)
+        ctx.invoke(upload_observations_to_loqusdb, case_id=case.internal_id)
+        ctx.invoke(upload_to_gens, case_id=case.internal_id)
+
+        # Clinical delivery upload
+        ctx.invoke(upload_clinical_delivery, case_id=case.internal_id)
+
+        LOG.info(
+            f"Upload of case {case.internal_id} was successful. Setting uploaded at to {dt.datetime.now()}"
```
Comment (henrikstranneheim) [inline cg/meta/upload/raredisease/raredisease.py:44]:
```suggestion
            f"Upload of case {case.internal_id} was successful. Updating uploaded at. in database"
```
To not leak implementation details.

File: cg/meta/workflow/analysis.py [hunk 8]
```diff
@@ -722,3 +724,24 @@ def get_data_analysis_type(self, case_id: str) -> str | None:
                 f"Case samples have different analysis types {', '.join(analysis_types)}"
             )
         return analysis_types.pop() if analysis_types else None
+
+    def get_bcf_file(self, hk_version_obj: Version) -> File:
+        """Fetch a bcf file and return the file object"""
```
Comment (henrikstranneheim) [inline cg/meta/workflow/analysis.py:729]:
```suggestion
        """Return a BCF file object."""
```

File: cg/meta/workflow/analysis.py [hunk 9]
```diff
@@ -722,3 +724,24 @@ def get_data_analysis_type(self, case_id: str) -> str | None:
...
+        genotype_files: list = self._get_genotype_files(version_id=hk_version_obj.id)
+        for genotype_file in genotype_files:
+            if self._is_variant_file(genotype_file=genotype_file):
+                LOG.debug(f"Found bcf file {genotype_file.full_path}")
```
Comment (henrikstranneheim) [inline cg/meta/workflow/analysis.py:733]:
```suggestion
                LOG.debug(f"Found BCF file {genotype_file.full_path}")
```

File: cg/meta/workflow/analysis.py [hunk 10]
```diff
@@ -722,3 +724,24 @@ def get_data_analysis_type(self, case_id: str) -> str | None:
...
+                return genotype_file
+        raise FileNotFoundError(f"No vcf or bcf file found for bundle {hk_version_obj.bundle_id}")
```
Comment (henrikstranneheim) [inline cg/meta/workflow/analysis.py:735]:
```suggestion
        raise FileNotFoundError(f"No VCF or BCF file found for bundle {hk_version_obj.bundle_id}")
```

File: cg/meta/workflow/analysis.py [hunk 11]
```diff
@@ -722,3 +724,24 @@ def get_data_analysis_type(self, case_id: str) -> str | None:
...
+
+    def _get_samples_sex(self, case_obj: Case, hk_version: Version) -> dict:
+        """Retrieve sex information from statusDB and from analysis prediction (stored HK QC metrics file)"""
```
Comment (henrikstranneheim) [inline cg/meta/workflow/analysis.py:738]:
```suggestion
        """Return sex information from StatusDB and from analysis prediction (stored Housekeeper QC metrics file)."""
```

File: cg/meta/workflow/analysis.py [hunk 12]
```diff
@@ -722,3 +724,24 @@ def get_data_analysis_type(self, case_id: str) -> str | None:
...
+        raise NotImplementedError
+
+    def get_qcmetrics_file(self, hk_version_obj: Version) -> Path:
+        """Fetch a qc_metrics file and return the path"""
```
Comment (henrikstranneheim) [inline cg/meta/workflow/analysis.py:742]:
```suggestion
        ""Return a QC metrics file path."""
```

File: cg/meta/workflow/analysis.py [hunk 13]
```diff
@@ -722,3 +724,24 @@ def get_data_analysis_type(self, case_id: str) -> str | None:
...
+        hk_qcmetrics = self.hk.files(
```
Comment (henrikstranneheim) [inline cg/meta/workflow/analysis.py:743]: Add type hint

File: cg/meta/workflow/analysis.py [hunk 14]
```diff
@@ -722,3 +724,24 @@ def get_data_analysis_type(self, case_id: str) -> str | None:
...
+            version=hk_version_obj.id, tags=HkAnalysisMetricsTag.QC_METRICS
+        ).first()
```
Comment (henrikstranneheim) [inline cg/meta/workflow/analysis.py:745]: Do we need to handle if there is none?

File: cg/meta/workflow/balsamic.py [hunk 15]
```diff
@@ -658,3 +658,16 @@ def is_analysis_normal_only(self, case_id: str) -> bool:
         if case.non_tumour_samples and not case.tumour_samples:
             return True
         return False
+
+    def _get_samples_sex(self, case_obj: Case) -> dict:
+        samples_sex = {}
```
Comment (henrikstranneheim) [inline cg/meta/workflow/balsamic.py:663]: Add doc str

File: cg/meta/workflow/balsamic.py [hunk 16]
```diff
@@ -658,3 +658,16 @@ def is_analysis_normal_only(self, case_id: str) -> bool:
...
+        for link_obj in case_obj.links:
```
Comment (henrikstranneheim) [inline cg/meta/workflow/balsamic.py:664]:
```suggestion
        for case_sample in case_obj.links:
```

File: cg/meta/workflow/balsamic.py [hunk 17]
```diff
@@ -658,3 +658,16 @@ def is_analysis_normal_only(self, case_id: str) -> bool:
         if case.non_tumour_samples and not case.tumour_samples:
             return True
         return False
+
+    def _get_samples_sex(self, case_obj: Case) -> dict:
```
Comment (henrikstranneheim) [inline cg/meta/workflow/balsamic.py:662]:
```suggestion
    def _get_samples_sex(self, case: Case) -> dict[str, str]:
```

File: cg/meta/workflow/balsamic.py [hunk 18]
```diff
@@ -658,3 +658,16 @@ def is_analysis_normal_only(self, case_id: str) -> bool:
...
+        samples_sex = {}
+        for link_obj in case_obj.links:
+            if link_obj.sample.is_tumour:
+                continue
+            sample_id = link_obj.sample.internal_id
+            samples_sex[sample_id] = {
```
Comment (henrikstranneheim) [inline cg/meta/workflow/balsamic.py:668]: Add tyoe hint

File: cg/meta/workflow/mip_dna.py [hunk 19]
```diff
@@ -94,3 +99,32 @@ def get_data_analysis_type(self, case_id: str) -> str:
             )
             return AnalysisType.WHOLE_GENOME_SEQUENCING
         return analysis_types.pop() if analysis_types else None
+
+    def _get_samples_sex(self, case_obj: Case, hk_version: Version) -> dict:
```
Comment (henrikstranneheim) [inline cg/meta/workflow/mip_dna.py:103]: Same changes here, see previous comments.

File: cg/meta/workflow/mip_dna.py [hunk 20]
```diff
@@ -94,3 +99,32 @@ def get_data_analysis_type(self, case_id: str) -> str:
...
+        qc_metrics_file = self.get_qcmetrics_file(hk_version)
+        analysis_sexes = self.analysis_sex(qc_metrics_file)
```
Comment (henrikstranneheim) [inline cg/meta/workflow/mip_dna.py:105]: Add type hints

File: cg/meta/workflow/mip_dna.py [hunk 21]
```diff
@@ -94,3 +99,32 @@ def get_data_analysis_type(self, case_id: str) -> str:
...
+        samples_sex = {}
+        for link_obj in case_obj.links:
+            sample_id = link_obj.sample.internal_id
+            samples_sex[sample_id] = {
+                "pedigree": link_obj.sample.sex,
+                "analysis": analysis_sexes[sample_id],
+            }
+        return samples_sex
+
+    def analysis_sex(self, qc_metrics_file: Path) -> dict:
```
Comment (henrikstranneheim) [inline cg/meta/workflow/mip_dna.py:115]:
```suggestion
    def get_analysis_sex(self, qc_metrics_file: Path) -> dict:
```
Add more details to return hint.

File: cg/meta/workflow/mip_dna.py [hunk 22]
```diff
@@ -94,3 +99,32 @@ def get_data_analysis_type(self, case_id: str) -> str:
...
+        """Fetch analysis sex for each sample of an analysis."""
```
Comment (henrikstranneheim) [inline cg/meta/workflow/mip_dna.py:116]:
```suggestion
        """Return analysis sex for each sample of an analysis."""
```
Use return with get_ in function name.

File: cg/meta/workflow/mip_dna.py [hunk 23]
```diff
@@ -94,3 +99,32 @@ def get_data_analysis_type(self, case_id: str) -> str:
...
+        qc_metrics: MIPMetricsDeliverables = self.get_parsed_qc_metrics_data(qc_metrics_file)
+        return {
+            sample_id_metric.sample_id: sample_id_metric.predicted_sex
+            for sample_id_metric in qc_metrics.sample_id_metrics
+        }
+
+    @staticmethod
+    def get_parsed_qc_metrics_data(qc_metrics: Path) -> MIPMetricsDeliverables:
+        """Parse the information from a qc metrics file"""
```
Comment (henrikstranneheim) [inline cg/meta/workflow/mip_dna.py:125]:
```suggestion
        ""Parse and return a QC metrics file."""
```

File: cg/meta/upload/gt.py [hunk 24]
```diff
@@ -44,74 +44,20 @@ def data(self, analysis: Analysis) -> dict:
...
-        ).first()
-        LOG.debug(f"Found qc metrics file {hk_qcmetrics.full_path}")
-        return Path(hk_qcmetrics.full_path)
-
-    @staticmethod
-    def get_parsed_qc_metrics_data(qc_metrics: Path) -> MIPMetricsDeliverables:
-        """Parse the information from a qc metrics file"""
-        qcmetrics_raw: dict = ReadFile.get_content_from_file(
-            file_format=FileFormat.YAML, file_path=qc_metrics
+        data["samples_sex"] = analysis_api._get_samples_sex(
```
Comment (henrikstranneheim) [inline cg/meta/upload/gt.py:57]: Is there a more descriptive name than data?

File: cg/meta/workflow/raredisease.py [hunk 25]
```diff
@@ -142,3 +147,33 @@ def get_workflow_metrics(self, sample_id: str) -> dict:
     @staticmethod
     def set_order_sex_for_sample(sample: Sample, metric_conditions: dict) -> None:
         metric_conditions["predicted_sex_sex_check"]["threshold"] = sample.sex
+
+    def _get_samples_sex(self, case_obj: Case, hk_version: Version) -> dict:
```
Comment (henrikstranneheim) [inline cg/meta/workflow/raredisease.py:151]: Same changes here

File: cg/meta/workflow/raredisease.py [hunk 26]
```diff
@@ -142,3 +147,33 @@ def get_workflow_metrics(self, sample_id: str) -> dict:
...
+        return samples_sex
+
+    def analysis_sex(self, qc_metrics_file: Path, sample_id: Sample) -> dict:
+        """Fetch analysis sex for each sample of an analysis."""
+        qc_metrics: list[MetricsBase] = self.get_parsed_qc_metrics_data(qc_metrics_file)
+        return str(
+            next(
+                metric.value
+                for metric in qc_metrics
+                if metric.name == "predicted_sex_sex_check" and metric.id == sample_id
```
Comment (henrikstranneheim) [inline cg/meta/workflow/raredisease.py:169]: It would be nice to get rid of these hard coded strs.

File: cg/meta/workflow/raredisease.py [hunk 27]
```diff
@@ -142,3 +147,33 @@ def get_workflow_metrics(self, sample_id: str) -> dict:
...
+        samples_sex = {}
+        for link_obj in case_obj.links:
+            sample_id = link_obj.sample.internal_id
+            samples_sex[sample_id] = {
+                "pedigree": link_obj.sample.sex,
+                "analysis": self.analysis_sex(qc_metrics_file, sample_id=sample_id),
+            }
+        return samples_sex
+
+    def analysis_sex(self, qc_metrics_file: Path, sample_id: Sample) -> dict:
```
Comment (henrikstranneheim) [inline cg/meta/workflow/raredisease.py:162]: Same changes as previous comment.

File: cg/meta/workflow/raredisease.py [hunk 28]
```diff
@@ -142,3 +147,33 @@ def get_workflow_metrics(self, sample_id: str) -> dict:
...
+        return str(
+            next(
+                metric.value
+                for metric in qc_metrics
+                if metric.name == "predicted_sex_sex_check" and metric.id == sample_id
+            )
+        )
+
+    @staticmethod
+    def get_parsed_qc_metrics_data(qc_metrics: Path) -> MetricsBase:
```
Comment (henrikstranneheim) [inline cg/meta/workflow/raredisease.py:174]: And here

File: tests/cli/upload/test_cli_upload_auto.py [hunk 29]
```diff
@@ -28,3 +28,24 @@ def test_upload_auto_with_workflow(
...
+
+
+def test_upload_auto_with_workflow_raredisease(
+    cli_runner: CliRunner,
+    helpers: StoreHelpers,
+    timestamp: datetime.datetime,
+    upload_context: CGConfig,
+    caplog,
+):
+    """Test upload auto"""
```
Comment (henrikstranneheim) [inline tests/cli/upload/test_cli_upload_auto.py:40]: Add more info

File: tests/cli/upload/test_cli_upload_auto.py [hunk 30]
```diff
@@ -28,3 +28,24 @@ def test_upload_auto_with_workflow(
...
+    # GIVEN a store with a workflow
+    workflow = Workflow.RAREDISEASE
+    helpers.add_analysis(store=upload_context.status_db, completed_at=timestamp, workflow=workflow)
+
+    # WHEN uploading all analysis from workflow MIP
```
Comment (henrikstranneheim) [inline tests/cli/upload/test_cli_upload_auto.py:45]:
```suggestion
    # WHEN uploading all analysis
```

File: tests/cli/upload/test_cli_upload_auto.py [hunk 31]
```diff
@@ -28,3 +28,24 @@ def test_upload_auto_with_workflow(
     cli_runner.invoke(upload_all_completed_analyses, ["--workflow", "mip-dna"], obj=upload_context)
     # THEN assert that the MIP analysis was successfully uploaded
     assert "Uploading analysis for case" in caplog.text
+
+
+def test_upload_auto_with_workflow_raredisease(
```
Comment (henrikstranneheim) [inline tests/cli/upload/test_cli_upload_auto.py:33]: Any chance of parametricizing the test?

File: tests/cli/upload/test_cli_upload_auto.py [hunk 32]
```diff
@@ -28,3 +28,24 @@ def test_upload_auto_with_workflow(
...
+):
+    """Test upload auto"""
+    # GIVEN a store with a workflow
+    workflow = Workflow.RAREDISEASE
+    helpers.add_analysis(store=upload_context.status_db, completed_at=timestamp, workflow=workflow)
+
+    # WHEN uploading all analysis from workflow MIP
+    caplog.set_level(logging.INFO)
+    cli_runner.invoke(
+        upload_all_completed_analyses, ["--workflow", "raredisease"], obj=upload_context
```
Comment (henrikstranneheim) [inline tests/cli/upload/test_cli_upload_auto.py:48]:
```suggestion
        upload_all_completed_analyses, ["--workflow", workflow], obj=upload_context
```

File: tests/cli/upload/test_cli_upload_observations.py [hunk 33]
```diff
@@ -107,3 +108,23 @@ def test_get_observations_api(cg_context: CGConfig, helpers: StoreHelpers):
...
+
+
+def test_get_observations_api_raredisease(cg_context: CGConfig, helpers: StoreHelpers):
+    """Test get observation API given a Loqusdb supported case."""
+    store: Store = cg_context.status_db
+
+    # GIVEN a Loqusdb supported case
+    case: Case = helpers.add_case(store, data_analysis=Workflow.RAREDISEASE)
+    sample: Sample = helpers.add_sample(store, application_type=SequencingMethod.WES)
+    link: CaseSample = store.relate_sample(case=case, sample=sample, status=PhenotypeStatus.UNKNOWN)
```
Comment (henrikstranneheim) [inline tests/cli/upload/test_cli_upload_observations.py:120]:
```suggestion
    case_sample: CaseSample = store.relate_sample(case=case, sample=sample, status=PhenotypeStatus.UNKNOWN)
```

**Review by henrikstranneheim — COMMENTED** [review]
Massive effort! Well done. 👍

File: cg/constants/observations.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jemten) [reply cg/constants/observations.py:56]: This tag should correspond to the normalized family vcf file coming from deepvariant, so it should be correspond to a unannotated, non-filtered snv researc vcf.

File: cg/constants/observations.py [hunk 34]
```diff
@@ -50,6 +50,14 @@ class MipDNAObservationsAnalysisTag(StrEnum):
...
+    SV_VCF: str = "vcf-sv-research"
```
[the remark this answers is not in the document]
Comment (jemten) [reply cg/constants/observations.py:57]: Here we're thinking of shifting to use the tag `vcf-sv` which should be the family sv vcf coming directly from svdb_merge.

File: cg/meta/report/raredisease.py
Comment (rannick) [reply cg/meta/report/raredisease.py:1]: Rebased

**Comment by ivadym** [comment]
FYI @rannick @peterpru: I've opened some issues to keep track of the missing files for the Scout upload and delivery report:

- https://github.com/Clinical-Genomics/cg/issues/3470
- https://github.com/Clinical-Genomics/cg/issues/3471

Additionally, the metrics reported and defined in [Pull Request 3459](https://github.com/Clinical-Genomics/cg/pull/3459) should be revised. For example, to calculate the mapped reads, I used `sample_metrics.mapped_reads / sample_metrics.total_reads` from the `metric_deliverables.yaml` file.

File: tests/fixture_plugins/observations_fixtures/observations_input_files_fixtures.py [hunk 35]
```diff
@@ -46,3 +47,22 @@ def mip_dna_observations_input_files(
...
+    return {
+        "family_ped_path": filled_file,
+        "profile_vcf_path": filled_file,
+        "snv_vcf_path": filled_file,
+        "sv_vcf_path": None,
+    }
+
+
+@pytest.fixture
+def raredisease_observations_input_files(
```
[the remark this answers is not in the document]
Comment (rannick) [reply tests/fixture_plugins/observations_fixtures/observations_input_files_fixtures.py:66]: That is the WES

**Comment by rannick** [comment]
Closing this. The base was not master so the commits got mixed up
